src/pipdeptree/_models/dag.py

```diff
@@ -110,25 +111,40 @@ def filter_nodes(self, include: set[str] | None, exclude: set[str] | None) -> Pa
         # documentation, `key` is simply
         # `project_name.lower()`. Refer:
         # https://setuptools.readthedocs.io/en/latest/pkg_resources.html#distribution-objects
+        include_with_casing_preserved: list[str] = []
         if include:
-            include = {s.lower() for s in include}
+            include_with_casing_preserved = include
+            include = [s.lower() for s in include]
         exclude = {s.lower() for s in exclude} if exclude else set()
 
         # Check for mutual exclusion of show_only and exclude sets
         # after normalizing the values to lowercase
         if include and exclude:
-            assert not (include & exclude)
+            assert not (set(include) & exclude)
```

Comment:
A tad off-topic, but do we need these lines? `--package` and `--exclude` aren't able to be used together anyways so there should never be a case where `include` and `exclude` will both be non-empty iterables

Reply:
Oh, after reading the CI logs I believe the CI failures (at least for testing) have to do with these lines since we have no tests that cover this code path (specifically we don't cover line 123). Not sure how there hasn't been complaints in previous commits to main though.

Reply:
I think is that all new changes must have coverage... even if they did not have before.

Reply:
Should I remove these lines (and maybe raise a `ValueError` as well as document an invariant explaining that `include` and `exclude` cannot both have values) or would you rather I add a test case to cover them?

Reply:
Add a test to cover them 👍